homeassistant/helpers/trigger.py

```diff
@@ -310,8 +311,16 @@ async def async_initialize_triggers(
     triggers: list[asyncio.Task[CALLBACK_TYPE]] = []
     for idx, conf in enumerate(trigger_config):
         # Skip triggers that are not enabled
-        if not conf.get(CONF_ENABLED, True):
-            continue
+        if CONF_ENABLED in conf:
+            enabled = conf[CONF_ENABLED]
+            if isinstance(enabled, Template):
+                try:
+                    enabled = enabled.async_render(variables, limited=True)
+                except TemplateError as err:
+                    log_cb(logging.ERROR, f"Error rendering enabled template: {err}")
+                    return None
```

Comment:
Why do we log and return `None`? Shouldn't we allow the error to bubble up instead?

Reply:
I did copy the behaviour of other errors in triggers here. 
See https://github.com/home-assistant/core/blob/e16a88a9c9850efb1ee2b684eb3e261e85906d93/homeassistant/helpers/trigger.py#L342-L352
Also, I didn't see any exception handling for `_async_attach_triggers` here
https://github.com/home-assistant/core/blob/e16a88a9c9850efb1ee2b684eb3e261e85906d93/homeassistant/components/automation/__init__.py#L843-L848

Would you know which error to raise to be handled properly?

Reply:
OK, you're right that letting the error bubble up may not work.

The existing behavior is to log when a trigger fails to set up. The behavior with this PR is that no trigger is set up if the `enabled` template can't render. Instead of that, log, and then `continue` with the next trigger.

In a separate PR, the function should let errors bubble up instead so we can better notify the user about the error when setting up the automation.

Reply:
Oh, you are right, I totally missed that. Fixed in https://github.com/home-assistant/core/pull/114458/commits/946ab1c1c0d696eadb0c00193681935f05bd9074